src/app/skill-groups/skill-groups.service.spec.ts

```diff
@@ -21,6 +23,157 @@ describe('SkillGroupsService', () => {
+      expect(group).toEqual(expectedResponse);
+    });
+
+    const request = httpTestingController.expectOne((req) =>
+      req.method === 'GET'
+      && req.url === `${environment.serverApiUrl}/groups/${expectedResponse.id}`
+    );
+
+    expect(request.request.responseType).toEqual('json');
+
+    request.flush(expectedResponse);
+  });
+
+  it('should create skill group', () => {
+    const expectedResponse: SkillGroup = {
+      id: '123123',
+      name: 'Skill group 1',
+      description: 'Skill group description'
+    };
+
+    skillGroupsService.createSkillGroup(expectedResponse.name, expectedResponse.description).subscribe((group) => {
+      expect(group).toEqual(expectedResponse);
+    });
+
+    const request = httpTestingController.expectOne((req) =>
+      req.method === 'POST'
+      && req.url === `${environment.serverApiUrl}/groups`
+    );
+
+    expect(request.request.responseType).toEqual('json');
+
+    request.flush(expectedResponse);
+  });
+
+  it('should update skill group', () => {
+    const expectedResponse: SkillGroup = {
+      id: '123123',
+      name: 'Skill group 1',
+      description: 'Skill group description'
+    };
+
+    skillGroupsService.updateSkillGroup(expectedResponse.id, expectedResponse.name, expectedResponse.description).subscribe((group) => {
+      expect(group).toEqual(expectedResponse);
+    });
+
+    const request = httpTestingController.expectOne((req) =>
+      req.method === 'PUT'
+      && req.url === `${environment.serverApiUrl}/groups/${expectedResponse.id}`
+    );
+
+    expect(request.request.responseType).toEqual('json');
+
+    request.flush(expectedResponse);
+  });
+
+  it('should delete the group', async(() => {
+    const groupId = 'd11235de-f13e-4fd6-b5d6-9c4c4e18aa4f';
+    skillGroupsService.deleteSkillGroup(groupId).subscribe((data: any) => {
+      expect(data).toBe(groupId);
+    });
+
+    const req = httpTestingController.expectOne({
+      method: 'DELETE',
+      url: `${environment.serverApiUrl}/groups/${groupId}`
+    });
+    expect(req.request.method).toBe('DELETE');
+
+    req.flush(groupId);
+  }));
+
+  it('should not send request to check if the skill already exists for empty search string', () => {
+    const searchParam = '';
+    skillGroupsService.isSkillGroupExist(searchParam).subscribe(actualResponse => {
+      expect(actualResponse).toBeFalsy();
+    });
+
+    httpTestingController.expectNone( `${environment.serverApiUrl}/groups/group-existence?search=${searchParam}`);
+  });
+
+  it('should send a request to check if the skill already exists', () => {
+    const searchParam = 'group';
+
+    skillGroupsService.isSkillGroupExist(searchParam).subscribe(actualResponse => {
+      expect(actualResponse).toBeTruthy();
+    });
+
+    const req = httpTestingController.expectOne( `${environment.serverApiUrl}/groups/group-existence?search=${searchParam}`);
+    expect(req.request.method).toBe('GET');
+    req.flush(1);
```

Comment:
It is better to return `true` here instead of `1`.

Reply:
I thought about it. but there is a bug in angular, so i found this [workaround ](https://github.com/angular/angular/issues/20690#issuecomment-403485479)with `req.flush(1)`